pandas/core/internals/api.py

```diff
@@ -51,6 +52,12 @@ def make_block(
 
     values, dtype = extract_pandas_array(values, dtype, ndim)
 
+    if klass is ExtensionBlock and is_period_dtype(values.dtype):
```

Comment:
defer to @jbrockmendel but not sure this is the right place for this

Reply:
this is fine.

issue a deprecation warning?

Reply:
> issue a deprecation warning?

For the case we know about (pyarrow), this is already fixed in the latest release. For in general, we might want to raise a warning for any time that `klass` is specified? So that it always will be inferred from the values (which pyarrow already does by now), and we don't have this issue again when changing Block classes.

Giving the timing of the release, I think we can discuss that in a follow up